test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template

```diff
@@ -364,6 +364,155 @@
+            var av = SPECIES.loadMask(a, i);
+            var bv = SPECIES.loadMask(b, i);
+            var cv = av.eq(bv);
+            cv.intoArray(r, i);
+        }
+        assertArraysEquals(a, b, r, $vectorteststype$::beq);
+    }
+
+    @Test(dataProvider = "maskProvider")
+    static void maskHashCode$vectorteststype$SmokeTest(IntFunction<boolean[]> fa) {
+        boolean[] a = fa.apply(SPECIES.length());
+
+        for (int i = 0; i < a.length; i += SPECIES.length()) {
+            var vmask = SPECIES.loadMask(a, i);
+            int hash = vmask.hashCode();
+
+            boolean subarr[] = Arrays.copyOfRange(a, i, i + SPECIES.length());
+            int expectedHash = Objects.hash(SPECIES, Arrays.hashCode(subarr));
+            Assert.assertTrue(hash == expectedHash, "at index " + i + ", hash should be = " + expectedHash + ", but is = " + hash);
+        }
+    }
+
+    @Test(dataProvider = "maskProvider")
+    static void maskTrueCount$vectorteststype$SmokeTest(IntFunction<boolean[]> fa) {
+        boolean[] a = fa.apply(SPECIES.length());
+
+        for (int i = 0; i < a.length; i += SPECIES.length()) {
+            var vmask = SPECIES.loadMask(a, i);
+            int tcount = vmask.trueCount();
+#if[!MaxBit]
+            int expectedTcount = Long.bitCount(vmask.toLong());
+#else[!MaxBit]
+            int expectedTcount = 0;
+            for (int j = i; j < i + SPECIES.length(); j++) {
+                expectedTcount += a[j] ? 1 : 0;
+            }
+#end[!MaxBit]
+
+            Assert.assertTrue(tcount == expectedTcount, "at index " + i + ", trueCount should be = " + expectedTcount + ", but is = " + tcount);
+        }
+    }
+
+    @Test(dataProvider = "maskProvider")
+    static void maskLastTrue$vectorteststype$SmokeTest(IntFunction<boolean[]> fa) {
+        boolean[] a = fa.apply(SPECIES.length());
+
+        for (int i = 0; i < a.length; i += SPECIES.length()) {
+            var vmask = SPECIES.loadMask(a, i);
+            int ltrue = vmask.lastTrue();
+            int j = i + SPECIES.length() - 1;
+            for (; j >= i; j--) {
+                if (a[j]) break;
+            }
+            int expectedLtrue = j - i;
+
+            Assert.assertTrue(ltrue == expectedLtrue, "at index " + i +
+                ", lastTrue should be = " + expectedLtrue + ", but is = " + ltrue);
+        }
+    }
+
+#if[!MaxBit]
+    @DataProvider
+    public static Object[][] longMaskProvider() {
+        return new Object[][]{
+                {0xFFFFFFFFFFFFFFFFl},
+                {0x0000000000000000l},
+                {0x5555555555555555l},
+                {0x0123456789abcdefl},
+        };
+    }
+
+    @Test(dataProvider = "longMaskProvider")
+    static void maskFromToLong$vectorteststype$SmokeTest(long inputLong) {
+        var vmask = VectorMask.fromLong(SPECIES, inputLong);
+        long outputLong = vmask.toLong();
+        Assert.assertEquals(outputLong, inputLong & (((1l << (SPECIES.length()-1)) << 1)-1));
```

Comment:
Recommend using `L` for the identifier of a long value (see also array declaration in `longMaskProvider`)
since lower case letter `l` is hard to distinguish from the digit `1`, and placing spaces in the subtracting expression. Alternatively could use `SPECIES.maskAll(true).toLong()` to avoid a complex expression dealing with the edge case of 64 lanes.

Reply:
Added L and missing spaces. Kept the complex expression as the test itself is for fromLong and toLong so using toLong in expectedValue may not be preferable.